src/hardware/pounder/dds_output.rs

```diff
@@ -113,7 +113,8 @@ impl DdsOutput {
         // fashion.
         let regs = unsafe { &*hal::stm32::QUADSPI::ptr() };
 
-        if regs.sr.read().flevel() != 0 {
+        // Warn if the fifo is still at least half full.
+        if regs.sr.read().flevel().bits() >= 16 {
```

Comment:
16 bytes is indeed half of the FIFO, which docs say is 32 bytes wide. However, thoughout this file, there are references to the FIFO being 4 32-bit words, which is only 16 bytes. Can you please correct the references in this file to reference 8 32-bit words wide, or 32 bytes?

Reply:
done